labman/gui/handlers/process_handlers/pooling_process.py

```diff
@@ -7,12 +7,155 @@
+                                ('min_val', 'min-val-'),
+                                ('max_val', 'max-val-'),
+                                ('blanks', 'blank-val-')]}}
+
+
+class LibraryPoolProcessHandler(BaseHandler):
+    @authenticated
+    def get(self):
+        plate_ids = self.get_arguments('plate_id')
+        self.render('library_pooling.html', plate_ids=plate_ids)
+
+    @authenticated
+    def post(self):
+        plate_ids = json_decode(self.get_argument('plate_ids'))
+        pool_base_name = self.get_argument('pool_base_name')
+        robot = Equipment(self.get_argument('robot'))
+        pool_volume = self.get_argument('pool_volume')
+        pool_f, pool_params = get_pool_params(self)
+        plate_pools = self.calculate_pools(plate_ids, pool_f, pool_params)
+        output_dict = {}
+        p_processes = []
+        for plate_id in plate_ids:
+            plate = Plate(plate_id)
+            pool_name = '%s_%s'.format(pool_base_name, plate.external_id)
+            # get pooling results for each plate
+            wells, conc_vals, pool_vols = plate_pools[plate_id]
+            # create input molar percentages
+            pcts = calc_pool_pcts(conc_vals, pool_vols)
+            q_process = plate.quantification_process
+            # create input compositions
+            comps = [{'composition': c,
+                      'input_volume': v,
+                      'percentage_of_output': p} for c, v, p in zip(wells,
+                                                                    pool_vols,
+                                                                    pcts)]
+            # create pooling process object for each pooled plate
+            p_process = PoolingProcess.create(
+                self.current_user, q_process, pool_name,
+                pool_volume, comps, robot)
+            # append to list of process ids
+            p_processes.append(p_process.id)
+        self.write({'processes': p_processes})
+
+
+# The ComputeLibraryPoolValueslHandler is meant to calculate the results from
+# the pooling process and disply for user approval.
+class ComputeLibraryPoolValueslHandler(BaseHandler):
+    @authenticated
+    def post(self):
+        plate_info = json_decode(self.get_argument('plate-info'))
+
+        plate_id = plate_info['plate-id']
+        func_name = plate_info['pool-func']
+        func_info = POOL_FUNCS[func_name]
+        function = func_info['function']
+        params = {arg: plate_info['%s%s' % (pfx, plate_id)]
+                  for arg, pfx in func_info['parameters']}
+
+        plate = Plate(plate_id)
+        quant_process = plate.quantification_process
+
+        output = {}
+        if func_name == 'Amplicon':
+            # Amplicon
+            # Compute the normalized concentrations
+            quant_process.compute_concentrations(**params)
+            # Compute the pooling values
+            output['pool_vals'] = make_2D_array(plate, quant_process)
```

Comment:
It doesn't appear that there is any pooling function being applied here.

Reply:
According to Greg, the normalized concentrations is what they as pooling values.